### PR TITLE
Make swift tests more stable with regard to leaking realms

### LINKED ISSUE #2120: Swift's testRefresh(), testAutorefresh() and testRealm() occasionally fail
By leaking realms: `RealmSwift/Tests/TestCase.swift:44: error: -[RealmSwift_Tests.RealmTests testRefresh] : XCTAssertNil failed: "Optional(<RLMRealm: 0x103f1d110>)"`. See https://ci.realm.io/job/objc_pr/1009/configuration=Debug,target=osx-swift/console.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #2120.

\c @jpsim @tgoyne

### CODE REVIEW

File: RealmSwift/Tests/RealmTests.swift [hunk 1]
```diff
@@ -489,11 +493,13 @@ class RealmTests: TestCase {
...
 
         let fired = expectationWithDescription("fired")
-        dispatch_async(dispatch_queue_create("background", nil)) {
+        let queue = dispatch_queue_create("background", nil)
+        dispatch_async(queue) {
             let otherThreadRealm = Realm()
             XCTAssertFalse(realm == otherThreadRealm)
             fired.fulfill()
         }
         waitForExpectationsWithTimeout(2, handler: nil)
```
Comment (tgoyne) [inline RealmSwift/Tests/RealmTests.swift]: The expectation here is pointless with the addition of the `dispatch_sync()` (in the other places it's needed to give the notification block a chance to run).

**Comment by tgoyne** [comment]
The pattern of dispatch_async() then immediately dispatch_sync() the same queue occurs enough that it's probably worth adding a `callOnDifferentThreadAndWait()` helper (preferably with a better name).

File: RealmSwift/Tests/RealmTests.swift (hunk 1 above)
Comment (segiddins) [inline RealmSwift/Tests/RealmTests.swift]: not pointless - it's necessary to check that the expectation has been fulfilled

Comment (tgoyne) [inline RealmSwift/Tests/RealmTests.swift]: That's not a thing that needs to be done. The code was previously just using the expectation to wait for the background block to finish.

Comment (segiddins) [inline RealmSwift/Tests/RealmTests.swift]: (you're right that it's pointless in this case, but it wasn't in the other 2)

Comment (tgoyne) [inline RealmSwift/Tests/RealmTests.swift]: Yes, that's exactly what I said.

**Comment by jpsim** [comment]
Looks good!

**Pull request merged**